datastream-server/src/main/java/com/linkedin/datastream/server/assignment/StickyPartitionAssignmentStrategy.java

```diff
@@ -592,4 +639,39 @@ private int getNumTasksForDatastreamFromZK(String stream) {
 
     return Integer.parseInt(_zkClient.readData(numTasksPath));
   }
+
+  /**
+   * Register metrics related to elastic task assignment
+   */
+  private void registerElasticTaskAssignmentMetrics(String taskPrefix) {
+    Supplier<Double> actualPartitionsPerTaskSupplier = () ->
+        (double) (_elasticTaskAssignmentInfoHashMap.containsKey(taskPrefix)
+            ? _elasticTaskAssignmentInfoHashMap.get(taskPrefix).getActualPartitionsPerTask() : 0);
+    DYNAMIC_METRICS_MANAGER.registerGauge(CLASS_NAME, taskPrefix, ACTUAL_PARTITIONS_PER_TASK,
+        actualPartitionsPerTaskSupplier);
+
+    Supplier<Double> needsAdjustmentSupplier = () ->
+        (_elasticTaskAssignmentInfoHashMap.containsKey(taskPrefix)
+            && _elasticTaskAssignmentInfoHashMap.get(taskPrefix).getNeedsAdjustment()) ? 1.0 : 0.0;
+    DYNAMIC_METRICS_MANAGER.registerGauge(CLASS_NAME, taskPrefix, PARTITIONS_PER_TASK_NEEDS_ADJUSTMENT,
+        needsAdjustmentSupplier);
+  }
+
+  private static class ElasticTaskAssignmentInfo {
+    private int _actualPartitionsPerTask;
+    private boolean _needsAdjustment;
+
+    ElasticTaskAssignmentInfo(int actualPartitionsPerTask, boolean needsAdjustment) {
```

Comment:
[FYI: Copying this comment I got from Ahmed once for inner classes]

Relevant: I remember looking at [this](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fstackoverflow.com%2Fquestions%2F6264657%2Fwhy-make-private-inner-class-member-public-in-java&data=02%7C01%7Cvmaheshwari%40linkedin.com%7C93a5aac105654d87be6908d7ed448b1c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637238750965657244&sdata=MZSudl8GkJj0yCq7ATaRGzwUUrysy8w4GOo%2BUsR%2Bftk%3D&reserved=0) SO question a long while ago, and liking this [answer](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fstackoverflow.com%2Fquestions%2F6264657%2Fwhy-make-private-inner-class-member-public-in-java&data=02%7C01%7Cvmaheshwari%40linkedin.com%7C93a5aac105654d87be6908d7ed448b1c%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637238750965667242&sdata=Nj3VR5a5gF0Rz7uM7k73o42bDVvI8Dt4HuWs1jShI4w%3D&reserved=0):

This method can be made public in order to indicate that it's semantically public, despite the fact that compiler doesn't enforce visibility rules in this particular case.

Imagine that during some refactoring you need to make this inner class top-level. If this method is private, how would you decide whether it should be made public, or some more restrictive modifier should be used? Declaring method as public tells reader the intentions of original author - this method shouldn't be considered an implementation detail.

It's worth noting the top-voted answer said using public was "nonsense" (i.e. ineffective/inconsequential) nonetheless.

Reply:
So are you suggesting I make this class public?

Reply:
done